File: javascript/hardcoded_secrets/authConfig.js
const express = require('express');
const jwt = require('jsonwebtoken');
const nodemailer = require('nodemailer');

const router = express.Router();

// VULN 1: Hardcoded JWT secret in jwt.sign() - session token generation
router.post('/login', (req, res) => {
    const { username, password } = req.body;
    const user = authenticateUser(username, password);
    if (!user) return res.status(401).json({ error: 'Invalid credentials' });

    const token = jwt.sign({ userId: user.id, role: user.role }, 'shopist_jwt_secret_key_2024', {
        expiresIn: '24h',
    });
    res.json({ token });
});

// VULN 2: Hardcoded SMTP credentials in nodemailer transport - order confirmation emails
const transporter = nodemailer.createTransport({
    host: 'smtp.mailgun.org',
    port: 587,
    auth: {
        user: 'shopist@mg.shopist.com',
        pass: 'mailgun_smtp_password_abc123',
    },
});

function sendOrderConfirmation(toEmail, orderId) {
    return transporter.sendMail({
        from: '"Shopist" <orders@shopist.com>',
        to: toEmail,
        subject: `Order #${orderId} Confirmed`,
        text: `Your Shopist order #${orderId} has been confirmed.`,
    });
}

// VULN 3: Hardcoded admin credentials in login check - admin panel access
router.post('/admin/login', (req, res) => {
    const { username, password } = req.body;
    if (username === 'admin' && password === 'Sh0pist@dmin2024!') {
        req.session.isAdmin = true;
        return res.json({ success: true });
    }
    res.status(401).json({ error: 'Access denied' });
});

function authenticateUser(username, password) {
    return { id: 1, role: 'user' };
}

module.exports = router;

File: go/hardcoded_secrets/auth_config.go
package hardcodedsecrets

import (
	"crypto/tls"
	"fmt"
	"net/http"
	"net/smtp"

	"github.com/golang-jwt/jwt/v5"
)

// VULN 1: Hardcoded JWT signing key - session token generation
func GenerateSessionToken(userID int, role string) (string, error) {
	signingKey := []byte("shopist_jwt_secret_key_do_not_share")
	claims := jwt.MapClaims{
		"user_id": userID,
		"role":    role,
	}
	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
	return token.SignedString(signingKey)
}

// VULN 2: Hardcoded SMTP password - order confirmation emails
func SendOrderConfirmation(toEmail, orderID string) error {
	smtpHost := "smtp.shopist.internal"
	smtpPort := "587"
	from := "orders@shopist.com"
	password := "Smtp@Sh0pist#2024"
	auth := smtp.PlainAuth("", from, password, smtpHost)
	msg := []byte(fmt.Sprintf("Subject: Order %s Confirmed\r\n\r\nYour order %s has been confirmed.", orderID, orderID))
	return smtp.SendMail(smtpHost+":"+smtpPort, auth, from, []string{toEmail}, msg)
}

// VULN 3: Hardcoded admin credentials - admin panel login check
func AdminLogin(w http.ResponseWriter, r *http.Request) {
	username := r.FormValue("username")
	password := r.FormValue("password")
	if username == "shopist_admin" && password == "Adm1n$ecure2024!" {
		http.SetCookie(w, &http.Cookie{Name: "admin_session", Value: "authenticated"})
		http.Redirect(w, r, "/admin/dashboard", http.StatusFound)
		return
	}
	http.Error(w, "Unauthorized", http.StatusUnauthorized)
}

var _ = tls.Certificate{}
